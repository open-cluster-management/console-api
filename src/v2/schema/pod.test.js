/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('Pods Resolver', () => {
  test('Correctly Resolves Pods Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          pods {
            cluster {
              metadata {
                name
              }
            }
            hostIP
            metadata {
              creationTimestamp
              labels
              name
              namespace
              uid
            }
            podIP
            restarts
            startedAt
            status
            images
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));

  test('Correctly Resolves Pod log query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          logs(containerName: "search-api", podName:"search-prod-28a0e-search-api-66cf776db5-7bzfh", podNamespace:"open-cluster-management", clusterName:"cluster1")
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
