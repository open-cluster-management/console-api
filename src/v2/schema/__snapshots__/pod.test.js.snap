// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pods Resolver Correctly Resolves Pod log query 1`] = `
Object {
  "data": Object {
    "logs": "{\\"kind\\":\\"Status\\",\\"apiVersion\\":\\"v1\\",\\"metadata\\":{},\\"status\\":\\"Failure\\",\\"message\\":\\"managedclusterinfos.internal.open-cluster-management.io \\\\\\"cluster1\\\\\\" not found\\",\\"reason\\":\\"NotFound\\",\\"details\\":{\\"name\\":\\"cluster1\\",\\"group\\":\\"internal.open-cluster-management.io\\",\\"kind\\":\\"managedclusterinfos\\"},\\"code\\":404}
",
  },
}
`;

exports[`Pods Resolver Correctly Resolves Pods Query 1`] = `
Object {
  "data": Object {
    "pods": Array [],
  },
}
`;
